hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ExecutionService.java

```diff
@@ -262,8 +256,9 @@ private void stealWork() {
         private final AtomicInteger seq = new AtomicInteger();
 
         @Override
-        public Thread newThread(Runnable r) {
-            return createThread(r, "blocking", seq.getAndIncrement());
+        public Thread newThread(@Nonnull Runnable r) {
+            return new Thread(r,
+                    String.format("hz.%s.jet.blocking.thread-%d", hzInstanceName, seq.getAndIncrement()));
```

Comment:
It would be nice, if vertex name, processor class and processor instance could be added to thread name...